src/main/java/org/pesmypetcare/webservice/dao/WeekTrainingDao.java

```diff
@@ -0,0 +1,76 @@
+package org.pesmypetcare.webservice.dao;
+
+import org.pesmypetcare.webservice.entity.WeekTrainingEntity;
+import org.pesmypetcare.webservice.error.DatabaseAccessException;
+
+import java.util.List;
+import java.util.Map;
+
+public interface WeekTrainingDao {
+    /**
+     * Creates a weekTraining of the pet on the database.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param date Creation date of the pet's weekTraining
+     * @param weekTrainingEntity The parameters of weekTraining
+     */
+    void createWeekTraining(String owner, String petName, String date, WeekTrainingEntity weekTrainingEntity);
+
+    /**
+     * Deletes all weekTrainings of the pet.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    void deleteAllWeekTrainings(String owner, String petName) throws DatabaseAccessException;
+
+    /**
+     * Deletes the weekTraining with the specified pet and date.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param petDate Date of the weekTraining instance
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    void deleteWeekTrainingByDate(String owner, String petName, String petDate) throws DatabaseAccessException;
+
+    /**
+     * Get one weekTraining of the pet with the specified date.
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param petDate Date of the weekTraining instance
+     * @return The weekTrainingEntity with the data of weekTraining
+     * @throws DatabaseAccessException If an error occurs when accessing the database
+     */
+    WeekTrainingEntity getWeekTrainingByDate(String owner, String petName, String petDate) throws DatabaseAccessException;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 122).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5400494)